src/server/game/Spells/Spell.cpp

```diff
@@ -5484,9 +5484,16 @@ SpellCastResult Spell::CheckCast(bool strict)
             case SPELL_EFFECT_TALENT_SPEC_SELECT:
             {
                 ChrSpecializationEntry const* spec = sChrSpecializationStore.LookupEntry(m_misc.SpecializationId);
-                if (!spec || spec->ClassID != m_caster->getClass())
+                Player* player = m_caster->ToPlayer();
+
+                if (!spec || (spec->ClassID != m_caster->getClass() && !spec->IsPetSpecialization()))
                     return SPELL_FAILED_NO_SPEC;
 
+                if (spec->IsPetSpecialization())
+                    if (Pet* pet = player->GetPet())
+                        if (!pet || pet->getPetType() != HUNTER_PET || !pet->GetCharmInfo())
```

Comment:
Idk because on retail you can't cast specialization on pet if you don't have pet (grayed button on interface) so it's a satefycheck but maybe it's not necessary (but jackpoz i have close my old PR for this)